src/controllers/productController.go

```diff
@@ -86,6 +87,11 @@ func (controller *ProductController) Create(writer http.ResponseWriter, request
 		return
 	}
 
+	if updatedProduct == nil {
```

Comment:
Not really sure this use case would ever happen.  If there is an error it would be handled, otherwise it should return the new product.